src/engagement.ts

```diff
@@ -0,0 +1,535 @@
+    reactions: issue.reactions!,
+    comments_data: commentsData,
+    user: issue.user!,
+    assignees: issue.assignees!
+  }
+}
+
+/**
+ * Calculate engagement score for an issue based on the engagement algorithm
+ */
+export function calculateScore(issue: IssueDetails): number {
+  // Components based on C# reference implementation:
+  // - Number of Comments       => Indicates discussion and interest
+  // - Number of Reactions      => Shows emotional engagement
+  // - Number of Contributors   => Reflects the diversity of input
+  // - Time Since Last Activity => More recent activity indicates higher engagement
+  // - Issue Age                => Older issues might need more attention
+  // - Number of Linked PRs     => Shows active work on the issue (not implemented)
+
+  const totalComments = issue.comments
+  const totalReactions =
+    issue.reactions.total_count +
+    (issue.comments_data?.reduce((sum, comment) => sum + comment.reactions.total_count, 0) || 0)
+  const contributors = getUniqueContributors(issue)
+  const lastActivity = Math.max(1, getTimeSinceLastActivity(issue))
+  const issueAge = Math.max(1, getIssueAge(issue))
+  const linkedPullRequests = 0 // Not implemented yet
+
+  // Weights from C# implementation:
+  const CommentsWeight = 3
+  const ReactionsWeight = 1
+  const ContributorsWeight = 2
+  const LastActivityWeight = 1
+  const IssueAgeWeight = 1
+  const LinkedPullRequestsWeight = 2
+
+  const score =
+    CommentsWeight * totalComments +
+    ReactionsWeight * totalReactions +
+    ContributorsWeight * contributors +
+    LastActivityWeight * (1 / lastActivity) +
+    IssueAgeWeight * (1 / issueAge) +
+    LinkedPullRequestsWeight * linkedPullRequests
+
+  return Math.round(score)
+}
+
+/**
+ * Calculate previous score (7 days ago) based on C# reference implementation
+ */
+export async function calculatePreviousScore(issue: IssueDetails): Promise<number> {
+  const sevenDaysAgo = new Date(Date.now() - 7 * 24 * 60 * 60 * 1000)
+  const issueCreatedAt = new Date(issue.created_at)
+
+  // If issue was created less than 7 days ago, return 0
+  if (issueCreatedAt > sevenDaysAgo) {
+    return 0
+  }
+
+  // Create historic snapshot by filtering comments and reactions to 7 days ago
+  const historicComments =
+    issue.comments_data
+      ?.filter((comment) => {
+        const commentDate = new Date(comment.created_at)
+        return commentDate <= sevenDaysAgo
+      })
+      .map((comment) => ({
+        ...comment,
+        reactions: {
+          ...comment.reactions,
+          total_count: 0 // Simplified - would need to get historic reactions
+        }
+      })) || []
+
+  const historicIssue: IssueDetails = {
+    ...issue,
+    comments: historicComments.length,
+    comments_data: historicComments,
+    reactions: {
+      ...issue.reactions,
+      total_count: 0 // Simplified - would need to get historic reactions
+    },
+    updated_at: sevenDaysAgo.toISOString()
+  }
```

Comment:
Move this into a new function in `issue-details.ts` called `getHistoricalIssueDetails` so we can have the issue processing in a single place.